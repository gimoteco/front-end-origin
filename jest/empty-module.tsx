import * as React from 'react';
const SvgrMock = () => <span />;

export default SvgrMock;

export const ReactComponent = SvgrMock;
