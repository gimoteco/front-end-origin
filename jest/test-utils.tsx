import { render } from '@testing-library/react';
import * as React from 'react';
import { ThemeProvider } from 'styled-components';
import theme from '../src/styles/theme';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
