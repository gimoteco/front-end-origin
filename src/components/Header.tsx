import * as React from 'react';
import Fade from 'react-reveal/Fade';
import Logo from '../icons/logo.svg';
import * as Style from './Header.styles';

function Header() {
  return (
    <Style.HeaderWrapper>
      <Fade left>
        <Logo />
      </Fade>
    </Style.HeaderWrapper>
  );
}

export default React.memo(Header);
