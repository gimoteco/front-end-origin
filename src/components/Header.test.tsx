import React from 'react';
import { render } from '../../jest/test-utils';
import Header from './Header';

describe('Header', () => {
  it('should render according snapshot', () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });
});
