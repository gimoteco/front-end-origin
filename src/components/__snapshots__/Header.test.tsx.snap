// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render according snapshot 1`] = `
<div>
  <header
    class="sc-bdVaJa qMBHx"
  >
    <div
      class="react-reveal"
    >
      <span />
    </div>
  </header>
</div>
`;
